### DISCUSSION TITLE
Description for LYD_PARSE_NO_STATE  seem a little bit not clear

### DISCUSSION THREAD

**Discussion by lyzliyuzhi91** [CESNET/libyang#2019] [Q&A]
```
#define LYD_PARSE_NO_STATE  0x080000        /**< Forbid state data in the parsed data. Usually used with ::LYD_VALIDATE_NO_STATE. */
```
But In testcase `test_state`, I found that if I have used LYD_PARSE_NO_STATE , I actually don't need to use LYD_VALIDATE_NO_STATE anymore. But if I have not use LYD_PARSE_NO_STATE , i should use LYD_VALIDATE_NO_STATE  for such scene

**Comment by michalvasko** [comment]
It really depends on the use-case. I do not think using `LYD_VALIDATE_NO_STATE` makes sense for parsing when you should simply either use `LYD_PARSE_NO_STATE` or not. `LYD_VALIDATE_NO_STATE` can be used for separate validation of data. But in general, your observations are correct.

**Reply by lyzliyuzhi91** [reply]
Yes, so I think `Usually used with ::LYD_VALIDATE_NO_STATE` is a little bit misleading? Because they do not be used together actually.

**Reply by michalvasko** [reply]
Oh, I have not noticed the note there and looking at the description of `LYD_VALIDATE_NO_STATE`, they actually should be used together because it is not just about validation but also not adding implicit state data, which you almost certainly not want to be added if you use `LYD_PARSE_NO_STATE`.
